apps/settings/test/unit/panels/firefox_sync/panel_test.js

```diff
@@ -287,67 +299,159 @@ suite('Firefox Sync panel >', () => {
     });
   });
 
-  ['disabled',
-   'errored'].forEach(state => {
-    suite('onsyncchange "' + state + '"', () => {
-      var subject;
+  suite('onsyncchange "disabled"', () => {
+    var subject;
 
-      var listenerSpies = new Map();
+    var listenerSpies = new Map();
 
-      suiteSetup(() => {
-        subject = suiteSetupCommon();
-        setSubjectSpiesAndStubs(subject);
-        LOGGED_OUT_SCREEN_ELEMENTS.forEach(element => {
-          if (!element || !element.listener) {
-            return;
-          }
-          var spy = this.sinon.spy(subject, element.listener);
-          listenerSpies.set(element.name, spy);
-        });
-        subject.onsyncchange({
-          state: state
-        });
+    suiteSetup(() => {
+      subject = suiteSetupCommon();
+      setSubjectSpiesAndStubs(subject);
+      LOGGED_OUT_SCREEN_ELEMENTS.forEach(element => {
```

Comment:
Can the part that sets up and tears down listenerSpies be taken out as a function, there are a lot of repeating things here and below.